src/main/java/bio/terra/cli/apps/NextflowRunner.java

```diff
@@ -9,16 +12,27 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** This class contains logic related to running the supported tool nextflow. */
-public class NextflowHelper extends SupportedAppHelper {
-  private static final Logger logger = LoggerFactory.getLogger(NextflowHelper.class);
+/** This class contains logic related to running the supported app nextflow. */
+public class NextflowRunner implements Enable, Stop {
+  private static final Logger logger = LoggerFactory.getLogger(NextflowRunner.class);
 
   // Mount point for the nextflow sub-directory on the Docker container.
   public static final String NEXTFLOW_MOUNT_POINT = "/usr/local/etc/nextflow";
 
   // Default nextflow sub-directory relative to the directory where the Terra CLI is running.
+  // TODO: walk up the directory tree to find this directory relative to the top-level workspace
+  // dir, not the current dir
   private static final Path DEFAULT_NEXTFLOW_DIR = Paths.get(".", "nextflow");
 
+  private GlobalContext globalContext;
+  private WorkspaceContext workspaceContext;
+
+  /** Set the global and workspace context properties. */
+  public void setContext(GlobalContext globalContext, WorkspaceContext workspaceContext) {
```

Comment:
setContext was a method instead of part of the constructor so that it could be specified as part of the interface. Without the interfaces, agreed it works better in a constructor.

I think your comment about how to reflect what current state is required may still apply.
- I thought of the login flow as kind of user-facing input/output, and opted to call AuthenticationManager.loginUser (which launches the oauth flow) in the command classes.
- Many operations require a current user or workspace. I thought of this as a basic context check, and opted to call GlobalContext.requireCurrentTerraUser and WorkspaceContext.requireCurrentWorkspace in the methods where the relevant value will actually be used. This was more for debugging though -- it doesn't really help in terms of forcing you to call loginUser at the command level.

Anyway, that's what I was thinking when I set it up this way. Happy to continue discussion with potentially a follow-up PR to implement any improvements.

Reply:
Sounds good. I think we want to think more about forcing the right setup at the command level, but not now or in this PR.